sound/soc/sof/topology.c

```diff
@@ -1498,7 +1498,8 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	case snd_soc_dapm_kcontrol:
 	default:
 		dev_dbg(scomp->dev, "widget type %d name %s not handled\n", swidget->id, tw->name);
-		break;
+		kfree(swidget);
+		return ret;
```

Comment:
instead of returning `ret` we should return simply `0`?

In other PR you explicitly stated that if the widget is not used in pipeline then it is just ignored, yet it looks like that unused, virtual widgets got handled when tearing down a pipeline?

Why is this happening? If the not handled widget is not used then it should not mater if it is added or not as it is not part of any path, but if it needs to be part of the path then it can not be ignored.

Confusing.

Reply:
this is quite baffling for me too @ujfalusi . This code has been there for almost 2 years now in this state while we were happy thinking that we're ignoring the unhandled widgets. And why is this device crashing only after some recent unrelated changes is also even more surprising.